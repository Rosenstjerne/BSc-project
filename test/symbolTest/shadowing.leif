function int shadowed_function() {
    return 42;
}

function int test_function() {
    function int shadowed_function() {
        return 7;
    }
    return shadowed_function();
}

print(shadowed_function()); # Should print 42
print(test_function()); # Should print 7