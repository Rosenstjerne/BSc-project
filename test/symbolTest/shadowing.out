function null main() {
	function int shadowed_function() {
		return 42;
		
	}
	function int test_function() {
		function int shadowed_function() {
			return 7;
			
		}
		
		return shadowed_function();
		
	}
	
	print(shadowed_function());
	print(test_function());
	
}
